pkg/controller/deployment/deployment_controller.go

```diff
@@ -170,6 +174,25 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
+func removeSideCar(r *ReconcileDeployment, dep *appsv1.Deployment, ctx context.Context) {
+	jaegerInstance, hasLabel := dep.Labels[inject.Label]
+	if hasLabel {
```

Comment:
I feel that if this function has been called, we should remove the sidecar no matter the state of the label. For instance, we might have the label at the namespace level.